src/mbedtls_context.cpp

```diff
@@ -728,14 +728,16 @@ namespace sockpp {
             pinned_cert_validation_result_ = (crt->raw.len == pinned_cert_->raw.len &&
                                               0 == memcmp(crt->raw.p, pinned_cert_->raw.p, crt->raw.len));
         }
+		
+		auto &callback = get_auth_callback();
         
         if (depth == 0) { // leaf cert
             received_cert_data_ = string((const char *)crt->raw.p, crt->raw.len);
             
             int status = -1;
             if (pinned_cert_) {
                 status = pinned_cert_validation_result_;
-            } else if (auto &callback = get_auth_callback(); callback) {
+            } else if (callback) {
                 string certData((const char*)crt->raw.p, crt->raw.len);
                 status = callback(certData);
             }
```

Comment:
Don't we need separate status from the result of callback? If the result of callback is true, we clear the flags to 0, otherwise set the flag accordingly?

Reply:
Only leaf cert to be verified through the callback. When the callback is used, we don't care about the intermediates or root cert. Only the leaf cert is what is matter.

Reply:
Here we are checking the leaf cert (depth == 0). Callback returns true(1) or false(0). If its true, should we clear all flags, or just the ones after (&=) ? If status does not come from the callback, because of pinned cert, I don't know.

Reply:
okay

Reply:
Agreed with @borrrden on this.